mlir-sycl/lib/Conversion/SYCLToSPIRV/SYCLToSPIRV.cpp

```diff
@@ -180,6 +180,46 @@ template <typename OpTy> class GridOpPatternIndex : public GridOpPattern<OpTy> {
   }
 };
 
+template <unsigned Dimensions>
+static std::enable_if_t<(1 <= Dimensions && Dimensions < 4), int64_t>
+mirrorIndex(int64_t index);
+
+template <> int64_t mirrorIndex<1>(int64_t index) {
+  assert(index == 0 && "Invalid index");
+  return 0;
+}
+
+template <> int64_t mirrorIndex<2>(int64_t index) {
+  assert(index < 2 && "Invalid index");
+  return index == 0 ? 1 : 0;
+}
+
+template <> int64_t mirrorIndex<3>(int64_t index) {
+  switch (index) {
```

Comment:
Could alternatively use `return (2 - index)` if we test for correct range of `index`.